flow360/component/simulation/primitives.py

```diff
@@ -93,10 +93,73 @@ class ReferenceGeometry(Flow360BaseModel):
 
 
 class Transformation(Flow360BaseModel):
-    """Used in preprocess()/translator to meshing param for volume meshing interface"""
+    """Transformation that will be applied to a body group."""
 
-    axis_of_rotation: Optional[Axis] = pd.Field()
-    angle_of_rotation: Optional[AngleType] = pd.Field()
+    type_name: Literal["BodyGroupTransformation"] = pd.Field("BodyGroupTransformation", frozen=True)
+
+    origin: LengthType.Point = pd.Field(  # pylint:disable=no-member
+        (0, 0, 0) * u.m,  # pylint:disable=no-member
+        description="The origin for geometry transformation in the order of scale,"
+        " rotation and translation.",
+    )
+
+    axis_of_rotation: Axis = pd.Field((1, 0, 0))
+    angle_of_rotation: AngleType = pd.Field(0 * u.deg)  # pylint:disable=no-member
+
+    scale: list[pd.PositiveFloat] = pd.Field([1, 1, 1])
```

Comment:
I think tuple is better:
```suggestion
    scale: Tuple[pd.PositiveFloat, pd.PositiveFloat, pd.PositiveFloat] = pd.Field((1, 1, 1))
```

alternatively we had vector type before: https://github.com/flexcompute/Flow360/blob/9fbd0f8511586114fc5138b9be4c48b2c32286eb/flow360/component/types.py#L19
which accepts lists: [1,2,3]

Reply:
Changed to `Tuple[pd.PositiveFloat, pd.PositiveFloat, pd.PositiveFloat]`